README.md

```diff
@@ -51,82 +65,155 @@ The environment variable `DEBUG_ERROR` (boolean) controls whether or not errors
-
-`REDUX_LOG` and `REDUX_DEV_UI` are both turned on by default in `config/ui-debug.sh`.
+`REDUX_LOG` is turned on by default in `config/ui-debug.sh`.
 
 ### Local Development w/o Debug Mode(s)
 
 All debug options are turned *off* by default in `config/local.sh`.
 
 ## Tests
 
-There are two sets of (unit) tests for the code in this repository.
-
-The tests for all device and data-processing code currently run in the [nodejs](https://nodejs.org/en/) server-side JavaScript environment. (We plan to eventually migrate these tests to run in-browser since the code itself runs in-browser in the Chrome App.)
-
-The tests for all the UI code run using the [Karma test runner](https://karma-runner.github.io/0.13/index.html) in [the headless WebKit browser PhantomJS](http://phantomjs.org/) or the Chrome browser.
-
-To run the tests in this repository as they are run on Travis CI, use:
+To run the tests in this repository as they are run on CircleCI and Appveyor use:
 
 ```bash
 $ npm test
 ```
+or
+```bash
+$ yarn test
+```
 
-To run just the UI tests in both PhantomJS and Chrome *locally*, use:
+## Linting & Code Style
 
-```bash
-$ npm run browser-tests
+We use [ESLint](http://eslint.org/) to lint our JavaScript code. We try to use the same linting options across all our client apps, but there are a few exceptions in this application, noted with comments in the `.eslintrc` configuration file.
+
+To run the linter (which also runs on CircleCI and Appveyor with every push, along with `npm test`), use:
+
+```
+$ npm run lint
 ```
 
-To run just the device and data-processing tests in node, use:
+Aside from the (fairly minimal) JavaScript code style options we *enforce* through the linter, we ask that internal developers and external contributors try to match the style of the code in each module being modified. New modules should look to similar modules for style guidance. In React component code, use existing ES6/ES2015 components (not legacy ES5 components) as the style model for new components.
+
+**NB: Please keep ES5 and ES6/ES2015 code distinct. Do *NOT* use ES6/ES2105 features in ES5 modules (most easily recognizable by the use of `require` rather than `import`).**
+
+## Docs
+
+Docs reside in several places in this repository, such as `docs/` and `lib/drivers/docs`. They are built as a static site with [GitBook](https://www.gitbook.com/ 'GitBook') and served at [developer.tidepool.io/](http://developer.tidepool.io/) via [GitHub Pages](https://pages.github.com/ 'GitHub Pages').
+
+See [this guidance on our use of GitBook at Tidepool](http://developer.tidepool.io/docs/).
+
+## Publishing
+
+Release management and application updates are handled via the Github provider in the `electron-builder` project. The recommended workflow for a new production release is as follows:
+
+1. When you're working on what might become a new release, increment the version number in `package.json` and `app/package.json` and commit/push (on the branch)
```

Comment:
Just checked and that is indeed the case. I asked on the #electron-builder Slack channel, and the response was
>  in case of two package.json structure, root package.json should be marked as private, version not required here.

So it turns out if you run `npm version` in the `app/` directory, it will just bump the package.json there, nice. If you want it to also create a tag, you just do a `mkdir .git` in `app/` as a workaround first.

Advantages of using `npm version` includes the capability to add preversion and postversion scripts, and of course preventing typos. Disadvantages include not being able to specify `alpha` (I think), but it does have `prerelease` tags (e.g. v0.310.0-0).

Do we want to use it, or do you think it's better to bump the version manually?